pkg/agent/apiserver/handlers/ovstracing/handler.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovstracing
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"net/http"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/apiserver/handlers"
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsctl"
+)
+
+// Response is the response struct of ovsflows command.
+type Response struct {
+	Result string `json:"result,omitempty"`
+}
+
+type tracingPeer struct {
+	ovsPort string
+	// Name of a Pod or Service
+	name string
+	// Namespace of Pod or Service.
+	namespace string
+	ip        net.IP
+}
+
+type request struct {
+	inputPort   *tracingPeer
```

Comment:
it's a bit confusing to me, from a terminology perspective, that `inputPort` is of type `tracingPeer`, just like `source` and `destination`. It also seems that `ip` would not be a valid field for `inputPort`. Would you be ok with having a different type for `inputPort`, or do you think it would cause too much code duplication?

Reply:
Yes. It is just to share more code, so I used one struct for all three.

Reply:
that's fine by me, but maybe a comment would be nice for this field, saying that `ip` is invalid for `inputPort`, and it must be one of `namespace/name` or `ovsPort`

Reply:
Added comments.